acceptance/tests/agent/agent_disable_lockfile.rb

```diff
@@ -1,100 +1,83 @@
-        end
+        assert(agent.file_exist?(agent_disabled_lockfile),
+               "Failed to create disabled lock file '#{agent_disabled_lockfile}' on agent '#{agent}'")
+
         lock_file_content = file_contents(agent, agent_disabled_lockfile)
 
-        # This is a hack; we should parse the JSON into a hash, but I don't
-        # think I have a library available from the acceptance test framework
-        # that I can use to do that.  So I'm falling back to <gasp> regex.
-        lock_file_content_regex = /"disabled_message"\s*:\s*"#{expected_message}"/
-        unless lock_file_content =~ lock_file_content_regex
-          fail_test("Disabled lock file contents invalid; expected to match '#{lock_file_content_regex}', got '#{lock_file_content}' on agent '#{agent}'")
-        end
+        file_lock_json = JSON.parse(lock_file_content)
+        assert_equal(expected_message, file_lock_json["disabled_message"])
       end
-    end
 
-    step "attempt to run the agent (message: '#{expected_message}')" do
-      agents.each do |agent|
-        on(agent, puppet('agent', "--test --server #{master}"),
-                     :acceptable_exit_codes => [1]) do
+      step "attempt to run disabled agent (message: '#{expected_message}')" do
+        skip_test "unable to assert match in Japanese" if agent["locale"] == "ja"
+        on(agent, puppet("agent", "--test --server #{master}"),
+           acceptable_exit_codes: [1]) do |result|
           disabled_regex = /administratively disabled.*'#{expected_message}'/
-          unless result.stdout =~ disabled_regex
-            fail_test("Unexpected output from attempt to run agent disabled; expecting to match '#{disabled_regex}', got '#{result.stdout}' on agent '#{agent}'") unless agent['locale'] == 'ja'
-          end
+          assert_match(disabled_regex, result.stdout)
         end
       end
-    end
-
-    step "enable the agent (message: '#{expected_message}')" do
-      agents.each do |agent|
 
+      step "enable the agent (message: '#{expected_message}')" do
         agent_disabled_lockfile = "#{agent.puppet['vardir']}/state/agent_disabled.lock"
-        on(agent, puppet('agent', "--enable"))
-        if file_exists?(agent, agent_disabled_lockfile) then
-          fail_test("Failed to remove disabled lock file '#{agent_disabled_lockfile}' on agent '#{agent}'")
-        end
+        on(agent, puppet("agent", "--enable"))
+        assert_equal(agent.file_exist?(agent_disabled_lockfile), false,
+                     "Failed to remove disabled lock file '#{agent_disabled_lockfile}' on agent '#{agent}'")
       end
-    end
 
-    step "verify that we can run the agent (message: '#{expected_message}')" do
-      agents.each do |agent|
-        on(agent, puppet('agent', "--test --server #{master}"))
+      step "verify that we can run the agent (message: '#{expected_message}')" do
+        on(agent, puppet("agent", "--test --server #{master}"))
```

Comment:
If there isn't a master to connect to, then the agent run will fail and I think it will exit with 1, so I think we need to allow that exit code here, but not 0.

Reply:
Thanks for catching this @joshcooper .  I have updated the test accordingly.